### ISSUE TITLE
Netopeer GUI - Connection closed

### ISSUE THREAD

**Issue by chris7891** [CESNET/Netopeer-GUI#94]
Hello,
I have the following problem with the netopeer gui:
Login and connect are working without any problem, but the info box says: "Connection closed."  (refresh works!)
In Netopeer stands in the main window: "No results found."
However, the models are received correctly (server -> gui), among "ALL" modules are listed. 
If I try to commit any change the GUI crashes and I'll get linked to: `http: // localhost / sections / 1 / all /` 
If I try to add a new module (+ symbol) and click into the box I don’t get any choice of options.
The Netopeer server is running and works so far (tested with Demo GUI).
Do you know where the problem may be? Are there any folder permissions missing that are needed by the GUI?

I’d appreciate any help!! 
Thank you!

**Comment by alexadavid** [comment]
Hello, which version of NetopeerGUI are you using? The master one?

Does your server support get-schema operation? In DemoGUI, did you see the list of modules in upper navigation? 

For setting correct folder permissions you could use ```php app/console app:install --post=install```, which executes file mode changes. Internally, it runs this file https://github.com/CESNET/Netopeer-GUI/blob/master/src/FIT/NetopeerBundle/bin/netconfwebgui-postinstall.sh, which sets correct permissions for all files.

**Comment by chris7891** [comment]
Hello,
Thank you for your answer.
 Yes I use the Master one and in the DemoGui I can see all correct.

If I call your skript, the Gui isn't working, there is only a white screen. I had start this in the netopeer-gui folder.

I hope you can help me.
Following you can see the Screenshot of the first page.
![unbenannt](https://cloud.githubusercontent.com/assets/18329680/14491828/71a175fe-017b-11e6-9f8e-84edbac44539.JPG)

**Comment by alexadavid** [comment]
Hello, could you send me a log output, which is located in netopeergui_root_folder/app/logs/prod.data.log?

**Comment by chris7891** [comment]
Hello, 
Of course I can do it. Thank you for your help:)
[prod.data.log.zip](https://github.com/CESNET/Netopeer-GUI/files/227468/prod.data.log.zip)

**Comment by alexadavid** [comment]
It seems, like the models (schemas) did not (probably) convert correctly using pyang. Did you install pyang? Is it working correctly (as a standalone binary)? Get-schema operation seems to work good, so only the conversion to format for gui. If pyang is working correctly, new directories (named by random hash) should be generated in ```/var/www/netopeer-gui/src/FIT/NetopeerBundle/Models/../Data/models/```, for example for Nacm module is the directory ```/var/www/netopeer-gui/src/FIT/NetopeerBundle/Models/../Data/models/6d9e79b29834b0e045e20b9ffb2a57a6e869181e/ietf-netconf-acm/2012-02-22/```

Converted schema by pyang is necessary for all modules in GUI except of "all" section, which does not require schema information at all.

**Comment by chris7891** [comment]
I had Install pyang version 1.6. 
I found this under /var/www/netopeer-gui/src/FIT/NetopeerBundle/Data/models and not in the directory
/var/www/netopeer-gui/src/FIT/NetopeerBundle/Models/../Data. this doesn't exist on my system and on the demo system. But you are right on my system in  /var/www/netopeer-gui/src/FIT/NetopeerBundle/Data/models wasn't something.
But i copied that in this directory from the demo-version and it doesn't worked.
than I have an other question. I thought the modules are important for the server and not for the GUI? 
I thought the workflow is that you can connect over the GUI with different netopeer server and every server have different models and the GUI get this information seperatly from the netopeer server.
I checked :
- the connection from the system GUI to the demo sever        : problem
- the connection from the system GUI to the system server     :problem
- the connection from the demo GUI to the system server        :no problem
- the connection from the demo GUI to the demo server          :+1:  no problem

It can not have a connection between the models and the GUI

Am I wrong in my deliberations?

with best regards and thanks

**Comment by alexadavid** [comment]
Hello, the GUI shows modules as separate "tabs" in top nav bar, which is working in your case. The module from tab is in fact the "filter query" for netconf operations. But every module needs to have implemented get-schema operation. There is no problem in connection to the bunch of different netopeer servers - every model (schema) is downloaded only once (for each combination of name, namespace and version of module - informations from server capabilities) and saved into Data/models folder. The schema is checked for every new connection to any server and downloaded if missing in Data/models folder. 

For schema data, some of our custom operations are performed on this data - using pyang. 

Without get-schema operation support, only the "All" section is working correctly, because it does not provide GUI features based on schema data.

And for the problem of system GUI, could you send me a log again, please? Please, send me the date of your operation execution for better orientation in long log data.

Basically, there should be no difference in your system version and demo version.

**Comment by chris7891** [comment]
Hello,
here is the log again and I clean it befor I try to connect. Now It is only one Try for Connect, change Tab and refresh on the log.
[prod.data.log.zip](https://github.com/CESNET/Netopeer-GUI/files/248848/prod.data.log.zip)

with best regards and thanks
